test/test_torch.py

```diff
@@ -2175,41 +2175,72 @@ def test_lu(self):
-            # Warning related check
-            self.assertEqual(len(w), 1)
-            self.assertTrue("Passing RHS tensor with number of dimensions = 2 is deprecated" in str(w[-1].message))
-
         from common_utils import random_fullrank_matrix_distinct_singular_value
-        for batches in [2, 5]:
-            for k, n in zip([2, 3, 5], [3, 5, 7]):
-                A = cast(random_fullrank_matrix_distinct_singular_value(n, batches))
-                b = cast(torch.randn(batches, n, k))
-                LU_data, LU_pivots, infos = A.lu(get_infos=True)
-                self.assertEqual(infos, torch.zeros_like(infos))
-                x = torch.lu_solve(b, LU_data, LU_pivots)
-                b_ = torch.matmul(A, x)
-                self.assertEqual(b_, b)
+
+        for k, n in zip([2, 3, 5], [3, 5, 7]):
+            A = cast(random_fullrank_matrix_distinct_singular_value(n))
+            b = cast(torch.randn(n, k))
+            LU_data, LU_pivots, info = A.lu(get_infos=True, pivot=pivot)
+            self.assertEqual(info, torch.zeros_like(info))
+            x = torch.lu_solve(b, LU_data, LU_pivots)
+            b_ = torch.matmul(A, x)
+            self.assertEqual(b_, b)
 
     @skipIfNoLapack
     def test_lu_solve(self):
         self._test_lu_solve(self, lambda t: t)
 
+    @staticmethod
+    def _test_lu_solve_batched(self, cast, pivot=True):
+        from common_utils import random_fullrank_matrix_distinct_singular_value
+
+        def lu_solve_test_helper(A_dims, b_dims, cast, pivot):
+            A = cast(random_fullrank_matrix_distinct_singular_value(*A_dims))
+            LU_data, LU_pivots, infos = torch.lu(A, get_infos=True, pivot=pivot)
+            self.assertEqual(infos, torch.zeros_like(infos))
+            b = cast(torch.randn(*b_dims))
+            return A, LU_data, LU_pivots, b
+
+        # test against lu_solve: one batch with both choices of upper
+        A, LU_data, LU_pivots, b = lu_solve_test_helper((5, 1), (1, 5, 10), cast, pivot)
+        x_exp = torch.lu_solve(b.squeeze(0), LU_data.squeeze(0), LU_pivots.squeeze(0))
+        x = torch.lu_solve(b, LU_data, LU_pivots)
+        self.assertEqual(x, x_exp.unsqueeze(0))
+
+        # test against cholesky_solve in a loop: four batches with both choices of upper
+        A, LU_data, LU_pivots, b = lu_solve_test_helper((5, 4), (4, 5, 10), cast, pivot)
+        x_exp_list = []
+        for i in range(4):
+            x_exp = torch.lu_solve(b[i], LU_data[i], LU_pivots[i])
+            x_exp_list.append(x_exp)
+        x_exp = torch.stack(x_exp_list)
+
+        x = torch.lu_solve(b, LU_data, LU_pivots)
+        self.assertEqual(x, x_exp)
+
+        # basic correctness test
+        A, LU_data, LU_pivots, b = lu_solve_test_helper((5, 3), (3, 5, 10), cast, pivot)
+        x = torch.lu_solve(b, LU_data, LU_pivots)
+        self.assertLessEqual(b.dist(torch.matmul(A, x)), 1e-12)
+
+        # Test non-contiguous inputs.
+        if not TEST_NUMPY:
```

Comment:
I know that we do this a lot in our codebase, but it's nicer if we put the non-contiguous test into its own test case and use `unittest.skipIf(not TEST_NUMPY)` so we can tell on the CI if it it get skipped by accident

Reply:
Just like the casting behavior, I could send a follow-up PR to address this in other places as far as linear algebra tests go.